File: Core/src/main/java/com/antonchaynikov/core/data/location/LocationProviderModule.java
package com.antonchaynikov.core.data.location;

import android.content.Context;
import android.location.Location;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import dagger.Module;
import dagger.Provides;

@Module
public class LocationProviderModule {

    @Provides
    public LocationProvider provide(@NonNull Context context, @Nullable Filter<Location> locationFilter) {
        LocationProvider locationProvider = new LocationProviderPlatform(context);
        locationProvider.setFilter(locationFilter);
        return locationProvider;
    }

    @Provides
    public Filter<Location> provideLocationFilter() {
        return new LocationFilter();
    }
}

File: core/src/main/java/com/antonchaynikov/core/data/location/LocationProviderModule.kt
package com.antonchaynikov.core.data.location

import android.content.Context
import android.location.Location
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class LocationProviderModule {

    @Singleton
    @Provides
    fun provide(context: Context, locationFilter: Filter<Location>): LocationProvider {
        val locationProvider = LocationProviderImpl(context)
        locationProvider.setFilter(locationFilter)
        return locationProvider
    }

    @Provides
    fun provideLocationFilter(): Filter<Location> {
        return LocationFilter()
    }
}